### PR TITLE
Add onclick event callback

### PR DESCRIPTION
**Pull request by josh18**
Does this look ok? Let me know if you need me to change anything.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -134,8 +138,12 @@ class Dropzone extends React.Component {
   }
 
   open() {
+    const clickEvent = new MouseEvent('click', {
+      bubbles: false
+    });
+
     this.fileInputEl.value = null;
-    this.fileInputEl.click();
+    this.fileInputEl.dispatchEvent(clickEvent);
```
Comment (okonet) [inline src/index.js]: Why is that needed?

File: src/index.js [hunk 2]
```diff
@@ -119,10 +119,14 @@ class Dropzone extends React.Component {
...
   }
 
-  onClick() {
+  onClick(e) {
     if (!this.props.disableClick) {
       this.open();
     }
+
+    if (this.props.onClick) {
+      this.props.onClick.call(this, e);
```
Comment (okonet) [inline src/index.js]: Should probably also check if it is a `function` Here for better guarding.

**Comment by okonet** [comment]
Tests are failing on CI.

File: src/index.js (hunk 2 above)
Comment (josh18) [inline src/index.js]: I just copied the previous callbacks style, example :https://github.com/okonet/react-dropzone/blob/master/src/index.js#L53-L55

Also not sure if checking is a function is necessary due to it being checked in propTypes?

File: src/index.js (hunk 1 above)
Comment (josh18) [inline src/index.js]: Otherwise the click bubbles up and triggers the DropZone click event again. Example: https://jsfiddle.net/8c13nd2y/

**Comment by josh18** [comment]
Yeah I saw the tests are failing but I'm not sure why, [MouseEvent is a valid API](https://developer.mozilla.org/en/docs/Web/API/MouseEvent), maybe something needs to be added to `mocha-environment.js` to make it pass?

File: src/index.js (hunk 2 above)
Comment (okonet) [inline src/index.js]: Probably. Leave it for now. This needs to be refactored for all places. FIY propTypes does not guard in runtime, that's why I think it's a good idea to check the type.

File: src/index.js (hunk 1 above)
Comment (okonet) [inline src/index.js]: Why not just cancel bubbling in the event handler then? This looks like a hack to me.

**Comment by okonet** [comment]
Yeah, we are using jsdom and it doesn't have all browser APIs out of the box. You have to mock it I guess.

File: src/index.js (hunk 1 above)
Comment (josh18) [inline src/index.js]: You mean like adding `onClick: (e) => { e.stopPropagation(); }` here? https://github.com/okonet/react-dropzone/blob/master/src/index.js#L207-L214

Comment (okonet) [inline src/index.js]: Yes. This will do the same, isn't it?

Comment (josh18) [inline src/index.js]: Unfortunately not (that's what I tried first), the onClick that happens there is a synthetic React event so `e.stopPropagation` doesn't work on it. See: http://stackoverflow.com/a/24421834/1661462

**Comment by okonet** [comment]
Can you please enable this https://github.com/blog/2247-improving-collaboration-with-forks so I can play around with it?

**Comment by josh18** [comment]
It should already be enabled?

**Comment by okonet** [comment]
Closing it in favor of https://github.com/okonet/react-dropzone/commit/33813573 which is already been released in https://github.com/okonet/react-dropzone/releases/tag/v3.8.0
